python/adjoint/objective.py

```diff
@@ -11,6 +11,18 @@
 
 Grid = namedtuple("Grid", ["x", "y", "z", "w"])
 
+#3 possible components for E x n and H x n 
+#signs are handled in code
+EH_TRANSVERSE    = [ [mp.Hy, mp.Hz, mp.Ey, mp.Ez],
+                     [mp.Hz, mp.Hx, mp.Ez, mp.Ex],
+                     [mp.Hx, mp.Hy, mp.Ex, mp.Ey] ]
+
+#Holds the components for each current source
+#for the cases of x,y, and z normal vectors.
+#This is the same as swapping H and E in the above list
+JK_TRANSVERSE    = [ [mp.Ey, mp.Ez, mp.Hy, mp.Hz],
```

Comment:
I think the component directions need to be switched too, right? e.g.

```
[mp.Ey, mp.Ez, mp.Hy, mp.Hz]
```

should be 

```
[mp.Ez, mp.Ey, mp.Hz, mp.Hy]
```

Reply:
Great point! I believe this was causing the "Assertion 'changed_materials' failed." bug. New push coming shortly.